### PR TITLE
Convert CCT impacts to CCT Severity in CFamilyIssueToAnalysisIssueConverter

### LINKED ISSUE #4802: Convert CCT impacts to CCT Severity in CFamilyIssueToAnalysisIssueConverter

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
fixes #4802

### CODE REVIEW

File: src/Core/Analysis/CleanCodeTaxonomy.cs [hunk 1]
```diff
@@ -98,8 +101,8 @@ public enum SoftwareQuality
     /// </summary>
     public enum SoftwareQualitySeverity
     {
-        High,
+        Low,
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Core/Analysis/CleanCodeTaxonomy.cs:104]: This ordering is rather weak. Either add explicit values for enum or add a custom IComparer

File: src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs [hunk 2]
```diff
@@ -109,9 +116,17 @@ public IAnalysisIssue Convert(Message cFamilyIssue, string sqLanguage, ICFamilyR
...
+            var ruleMetaData = rulesConfiguration.RulesMetadata[cFamilyIssue.RuleKey];
+
             // Look up default severity and type
-            var defaultSeverity = rulesConfiguration.RulesMetadata[cFamilyIssue.RuleKey].DefaultSeverity;
-            var defaultType = rulesConfiguration.RulesMetadata[cFamilyIssue.RuleKey].Type;
+            var defaultSeverity = ruleMetaData.DefaultSeverity;
+            var defaultType = ruleMetaData.Type;
+            SoftwareQualitySeverity? highestSoftwareQualitySeverity = null;
+
+            if (connectedModeFeaturesConfiguration.IsNewCctAvailable() && ruleMetaData.Type != IssueType.SecurityHotspot)
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs:126]: could swap operands to possibly make 1 less server request

File: src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs [hunk 3]
```diff
@@ -109,9 +116,17 @@ public IAnalysisIssue Convert(Message cFamilyIssue, string sqLanguage, ICFamilyR
             Debug.Assert(cFamilyIssue.Column > 0 || cFamilyIssue.Column == 0);
             Debug.Assert(cFamilyIssue.EndColumn > 0 || cFamilyIssue.EndLine == 0);
 
+            var ruleMetaData = rulesConfiguration.RulesMetadata[cFamilyIssue.RuleKey];
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs:119]: is it possible here to get a key that's not present in the dictionary? I would leave a comment or use TryGet

File: src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs [hunk 4]
```diff
@@ -275,15 +296,22 @@ private string CalculateLineHash(MessagePart cFamilyIssueLocation, IReadOnlyDict
...
                 case IssueType.SecurityHotspot:
                     return AnalysisIssueType.SecurityHotspot;
+
                 default:
                     throw new ArgumentOutOfRangeException(nameof(issueType));
             }
         }
+
+        internal /* for testing */ static SoftwareQualitySeverity? GetHighestSoftwareQualitySeverity(RuleMetadata ruleMetadata)
+            => ruleMetadata.Code?.Impacts?.Count > 0 ? (SoftwareQualitySeverity?)ruleMetadata.Code.Impacts.Max(r => r.Value) : null;
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs:315]: See my comment on the enum about ordering

File: src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs [hunk 5]
```diff
@@ -176,14 +191,16 @@ private ITextDocument GetTextDocument(string filePath)
             IssueSeverity defaultSeverity,
             IssueType defaultType,
             IReadOnlyList<IAnalysisIssueFlow> flows,
-            IReadOnlyDictionary<string, ITextDocument> fileContents)
+            IReadOnlyDictionary<string, ITextDocument> fileContents,
+            SoftwareQualitySeverity? highestSoftwareQualitySeverity
+            )
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs:196]: )

File: src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs (hunk 3 above)
Comment (ugras-ergun-sonarsource) [reply src/Integration.Vsix/CFamily/CFamilyIssueToAnalysisIssueConverter.cs:119]: No, it's not possible.

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**